### PR TITLE
device id pretty print and listener support

### PR DESCRIPTION
**Pull request by dagar**
nsh> listener sensor_gyro

    TOPIC: sensor_gyro instance 0 #1
    sensor_gyro_s
        timestamp: 6166722830  (0.001146 seconds ago)
        error_count: 0
        device_id: 2360354 (Type: 0x24, SPI:4 (0x04))
        x: -0.0041
        y: 0.0070
        z: 0.0014
        integral_dt: 4009
        x_integral: -0.0000
        y_integral: 0.0000
        z_integral: 0.0000
        temperature: 72.7867
        scaling: 0.0011
        x_raw: 5
        y_raw: -6
        z_raw: 2

### CODE REVIEW

File: msg/tools/px_generate_uorb_topic_helper.py [hunk 1]
```diff
@@ -250,6 +250,10 @@ def print_field(field):
         print("if (message.timestamp != 0) {\n\t\tPX4_INFO_RAW(\"\\t" + field.name + \
             ": " + c_type + "  (%.6f seconds ago)\\n\", " + field_name + \
             ", hrt_elapsed_time(&message.timestamp) / 1e6);\n\t} else {\n\t\tPX4_INFO_RAW(\"\\n\");\n\t}" )
+    elif field.name == 'device_id':
```
Comment (bkueng) [inline msg/tools/px_generate_uorb_topic_helper.py]: Can you also check that the type is int32_t?

File: src/lib/drivers/device/Device.hpp [hunk 2]
```diff
@@ -184,6 +202,23 @@ class __EXPORT Device
 		uint32_t devid;
 	};
 
+	/**
+	 * Print decoded device id string to a buffer.
+	 *
+	 * @param buffer                        buffer to write to
+	 * @param length                        buffer length
+	 * @param id	                        The device id.
+	 * @param return                        number of bytes written
```
Comment (bkueng) [inline src/lib/drivers/device/Device.hpp]: The string is not always null-terminated. This should be handled and/or remarked.

Comment (dagar) [reply src/lib/drivers/device/Device.hpp]: Explicitly null terminated

**Comment by LorenzMeier** [comment]
Awesome, super helpful!

**Pull request merged**